src/server/models/jobs.py

```diff
@@ -273,8 +273,14 @@ def find_remote_node_for_job(app, job: V2JobRecordSchema) -> str:
         nodeStatus = remote_node.getStatus()
         if nodeStatus.ableToRunJobs and nodeStatus.nodeArch == job.arch:
             app.logger.info(f"Matching remote node: {xname}, current jobs on node: {nodeStatus.numCurrentJobs}")
+            
+            # -1 means no job information, make sure we don't prefer those nodes
+            numNodeJobs = nodeStatus.numCurrentJobs
+            if numNodeJobs == -1:
+                numNodeJobs = 10000
```

Comment:
So, is the idea that you give the node really high number of jobs, so that it is not preferred. The thinking being, "Oh, this guy is really busy, let's leave him alone"?

Did you want to use the best_node_job_count here instead of 10000? Oh, best_node_job_count is variable that gets reset as you go through the loop, and you just set it arbitrarily high to start with. Okay, I see where this is going.

Reply:
Yes - take it if nothing else is there, but if one has a well-established current number of known jobs running use that one instead.

The number of jobs is just for the 'poor man's load balancing', but don't disqualify a node on that basis.